app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CommentsController < ApplicationController
+  before_action :set_post, only: [:create, :destroy]
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Post.comments.build(comment_params)
```

Comment:
I think there's a problem with this code. `Post` is the class name and `.build` is an instance method.

Have you run this in the browser or in the console?

I think you first need to work on your routes. Your routes will help you formulate the right strategies in building your controllers. Step back a bit and plan out your application very well. For instance I would think about "if a user wanna sees user's post what would be the route?" => "/users/posts/:id.. That means in the `routes.rb` file:

```
resources :users do
   resources :posts
end
```
Let's remember the controller is associated with routes. And you have to understand completely your routing to make meaningful controller actions.